packages/injectable/src/dependency-injection-container/createContainer.js

```diff
@@ -282,15 +282,17 @@ export default (...listOfGetRequireContexts) => {
 };
 
 const autoRegisterInjectables = ({ getRequireContextForInjectables, di }) => {
-  const requireContextForInjectables = getRequireContextForInjectables();
-
-  pipeline(
-    requireContextForInjectables,
-    invoke('keys'),
-    map(requireContextForInjectables),
-    map('default'),
-    forEach(di.register),
-  );
+  const context = getRequireContextForInjectables();
+
+  for (const key of context.keys()) {
+    const injectable = context(key).default;
+
+    if (!injectable) {
+      throw new Error(`File ${key} does not have a default export.`)
+    }
+
+    di.register(injectable);
+  }
```

Comment:
Use functional convention even in this seemingly trivial case for:
1. Codebase convention
2. The usual functional gospel: readability, composability, refactorability, etc.
3. Discussion elsewhere

```suggestion
pipeline(
      requireContextForInjectables,
      map(getDefaultExport),
      tap(forEach(verifyInjectableSignature)),
      forEach(di.register),
);```